packages/browser/src/transports/base.ts

```diff
@@ -54,6 +65,63 @@ export abstract class BaseTransport implements Transport {
     return this._buffer.drain(timeout);
   }
 
+  /**
+   * @inheritDoc
+   */
+  public recordLostEvent(type: Outcome, category: SentryRequestType): void {
+    if (!this.options.sendClientReport) {
+      return;
+    }
+    const key = `${type}:${CATEGORY_MAPPING[category]}`;
+    logger.log(`Adding ${key} outcome`);
+    this._outcomes[key] = (this._outcomes[key] ?? 0) + 1;
+  }
+
+  /**
+   * Send outcomes as an envelope
+   */
+  protected _flushOutcomes(): void {
+    if (!this.options.sendClientReport) {
+      return;
+    }
+
+    if (!navigator || typeof navigator.sendBeacon !== 'function') {
+      logger.warn('Beacon API not available, skipping sending outcomes.');
```

Comment:
Should we just disable recording outcomes all together if the beacon API is not working? Do some kind of early check?

Reply:
We may want to have a fallback for this after this PoC